cloud/ec2.go

```diff
@@ -523,8 +523,86 @@ func (m *ec2Manager) getResources(ctx context.Context, h *host.Host) ([]string,
 	return resources, nil
 }
 
-// ModifyHost modifies a host according to the changes specified by a HostModifyOptions struct.
-func (m *ec2Manager) ModifyHost(ctx context.Context, h *host.Host, changes host.HostModifyOptions) error {
+// addTags adds or updates the specified tags in the client and db
+func (m *ec2Manager) addTags(ctx context.Context, h *host.Host, tags []host.Tag, resources []string) error {
+	createTagSlice := make([]*ec2.Tag, len(tags))
+	for i := range tags {
+		createTagSlice[i] = &ec2.Tag{Key: &tags[i].Key, Value: &tags[i].Value}
+	}
+	_, err := m.client.CreateTags(ctx, &ec2.CreateTagsInput{
+		Resources: aws.StringSlice(resources),
```

Comment:
Are these resources tracked in the host object at all? It would be nice to not have to pass these, but that may not be feasible

Reply:
There's a VolumeIDs field in the host document that the GetVolumeIDs client method simply returns if it's ever set. I made this method also cache the VolumeIDs when it gets them from EC2.

The resources slice is basically the host ID with the volume IDs appended, so I just moved getResources inside each helper function -- if volume IDs is cached, then it shouldn't make any unnecessary calls to the EC2 API.